core/src/main/scala/com/softwaremill/react/kafka2/Consumer.scala

```diff
@@ -75,7 +89,7 @@ object ManualCommitConsumer {
   }
 }
 
-class ManualCommitConsumer[K, V](consumerProvider: ConsumerProvider[K, V])
+class ManualCommitConsumer[K, V](consumerProvider: () => KafkaConsumer[K, V])
```

Comment:
There are several point why I chose this API:
- setup of `KafkaCosumer` is not just properties set. It is also offset setup, topic subscriptions and possibly manual partitions assignment. So it can not be setup in terms of just properties. My idea was to build another API to setup `KafkaConsumer`. This API may share reusable immutable parts.
- you can easily share `() => KafkaConsumer ` aswell
- easy mocking / API extension. In case of mocking or performing some setup actions that are out of scope you can use just manual setup like 
```scala
ManualCommitConsumer(() => {
  // just create whatever you want
})
```